File: chrome/android/java/src/org/chromium/chrome/browser/safe_browsing/FileTypePolicies.java
// Copyright 2019 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package org.chromium.chrome.browser.safe_browsing;

import org.chromium.base.annotations.JNINamespace;

/**
 * This class reports UMA values based on files' extensions.
 */
@JNINamespace("safe_browsing")
public final class FileTypePolicies {
    /**
     * @param path The file path.
     * @return The UMA value for the file.
     */
    public static int umaValueForFile(String path) {
        return nativeUmaValueForFile(path);
    }

    private static native int nativeUmaValueForFile(String path);
}

File: src/chrome/browser/safe_browsing/android/file_type_policies.cc
// Copyright 2019 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "chrome/common/safe_browsing/file_type_policies.h"

#include "base/android/jni_string.h"
#include "base/files/file_path.h"
#include "jni/FileTypePolicies_jni.h"

namespace safe_browsing {

static jint JNI_FileTypePolicies_UmaValueForFile(
    JNIEnv* env,
    const base::android::JavaParamRef<jstring>& path) {
  base::FilePath file_path(ConvertJavaStringToUTF8(env, path));
  return safe_browsing::FileTypePolicies::GetInstance()->UmaValueForFile(
      file_path);
}

}  // namespace safe_browsing
